sdk/src/main/java/com/oursky/authgear/latte/fragment/LatteAppLink.kt

```diff
@@ -0,0 +1,43 @@
+package com.oursky.authgear.latte.fragment
+
+import android.net.Uri
+import com.oursky.authgear.data.HttpClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.net.URL
+import java.nio.charset.StandardCharsets
+
+sealed class LatteAppLink {
+    abstract suspend fun handle(presenter: LattePresenter)
+
+    internal class ResetLink(val query: List<Pair<String, String>>) : LatteAppLink() {
+        override suspend fun handle(presenter: LattePresenter) {
+            val delegate = presenter.delegate ?: return
+            val handle = presenter.resetPassword(query)
+            delegate.showLatteFragment(handle.id, handle.fragment)
+            try {
+                handle.await()
+            } finally {
+                delegate.hideLatteFragment(handle.id)
+            }
+        }
+    }
+
+    internal class LoginLink(val uri: Uri) : LatteAppLink() {
+        override suspend fun handle(presenter: LattePresenter) {
+            return withContext(Dispatchers.IO) {
+                val url = URL(uri.toString())
+                HttpClient.fetch(url, "POST", emptyMap()) { conn ->
+                    conn.errorStream?.use {
+                        val responseString = String(it.readBytes(), StandardCharsets.UTF_8)
+                        HttpClient.throwErrorIfNeeded(conn, responseString)
+                    }
+                    conn.inputStream.use {
+                        val responseString = String(it.readBytes(), StandardCharsets.UTF_8)
+                        HttpClient.throwErrorIfNeeded(conn, responseString)
+                    }
+                }
+            }
```

Comment:
Seems now we have activity link handler and fragment link handler. Should we extract the common logic from them to ensure they are same in both implementation?

Reply:
I was thinking of deprecating the activity applink implementation and remove it later, or refactor activity applink to use fragment applink.

Reply:
Choose whichever approach that takes the least effort.

Reply:
I'll delete the activity applink and flows in another PR.